server/channels/app/custom_profile_attributes_test.go

```diff
@@ -0,0 +1,462 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetCPAField(t *testing.T) {
+	os.Setenv("MM_FEATUREFLAGS_CUSTOMPROFILEATTRIBUTES", "true")
+	defer os.Unsetenv("MM_FEATUREFLAGS_CUSTOMPROFILEATTRIBUTES")
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	cpaGroup, rErr := th.App.Srv().propertyService.RegisterPropertyGroup(model.CustomProfileAttributesPropertyGroupName)
+	require.NoError(t, rErr)
+
+	t.Run("should fail when getting a non-existent field", func(t *testing.T) {
+		field, err := th.App.GetCPAField(model.NewId())
+		require.NotNil(t, err)
+		require.Equal(t, "app.custom_profile_attributes.get_property_field.app_error", err.Id)
+		require.Empty(t, field)
+	})
+
+	t.Run("should fail when getting a field from a different group", func(t *testing.T) {
+		field := &model.PropertyField{
+			GroupID: model.NewId(),
+			Name:    model.NewId(),
+			Type:    model.PropertyFieldTypeText,
+		}
+		createdField, err := th.App.Srv().propertyService.CreatePropertyField(field)
+		require.NoError(t, err)
+
+		fetchedField, appErr := th.App.GetCPAField(createdField.ID)
+		require.NotNil(t, appErr)
+		require.Equal(t, "app.custom_profile_attributes.property_field_not_found.app_error", appErr.Id)
+		require.Empty(t, fetchedField)
+	})
+
+	t.Run("should get an existing CPA field", func(t *testing.T) {
+		field := &model.PropertyField{
+			GroupID: cpaGroup.ID,
+			Name:    "Test Field",
+			Type:    model.PropertyFieldTypeText,
+			Attrs:   map[string]any{"visibility": "hidden"},
+		}
+
+		createdField, err := th.App.CreateCPAField(field)
+		require.Nil(t, err)
+		require.NotEmpty(t, createdField.ID)
+
+		fetchedField, err := th.App.GetCPAField(createdField.ID)
+		require.Nil(t, err)
+		require.Equal(t, createdField.ID, fetchedField.ID)
+		require.Equal(t, "Test Field", fetchedField.Name)
+		require.Equal(t, map[string]any{"visibility": "hidden"}, fetchedField.Attrs)
+	})
+}
+
+func TestListCPAFields(t *testing.T) {
+	os.Setenv("MM_FEATUREFLAGS_CUSTOMPROFILEATTRIBUTES", "true")
+	defer os.Unsetenv("MM_FEATUREFLAGS_CUSTOMPROFILEATTRIBUTES")
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	if cpaGroupID == "" {
+		_, err := th.App.cpaGroupID()
+		require.NoError(t, err)
+	}
```

Comment:
Consistency nit: we call `cpaGroupID()` here, but in the previous test we directly call `RegisterPropertyGroup`. Should we do it the same way in both tests?

Reply:
Standardized `groupID` access on 93f2dd5f1f, PTAL